lib/spack/spack/cmd/git.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import argparse
+
+import llnl.util.tty as tty
+from llnl.util.filesystem import working_dir
+
+import spack.cmd
+import spack.util.git
+from spack.cmd import spack_is_git_repo
+
+description = "run git inside of the spack repo"
+section = "system"
+level = "short"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        "command", nargs=argparse.REMAINDER, help="git command to run within spack repo"
+    )
+
+
+def git(parser, args):
+    # make sure this is a git repo
+    if not spack_is_git_repo():
+        tty.die("This spack is not a git clone. Can't use 'spack git'")
+    git = spack.util.git.git(required=True)
+
+    # execute git within the local spack repository
+    with working_dir(spack.paths.prefix):
+        output = git(*args.command, output=str, fail_on_error=False)
+        print(output.strip())
```

Comment:
Why pass the output to a string, just to print it?

Reply:
I felt like a single line combination was a bit long,
```
print(git(*args.command, output=str, fail_on_error=False).strip())
```

although rewriting as the following is still pretty understandable,
```
print(git(*args.command, output=str, fail_on_error=False), end='')
```

Reply:
Why not just run git and *not* redirect the output?